unity_jobs_simulator/Assets/Scripts/Robot_Controller.cs

```diff
@@ -270,4 +285,27 @@ public void checkAngleErrors()
             }
         }
     }
+
+    public Vector3 getTCPTranspose()
+    {
+        int final = _Arms.Length - 1;
+        ArticulationBody last = _Arms[final];
+        Vector3 worldTranspose = last.transform.position;
+        // Vector3 worldTranspose = Transform.TransformVector(localTranspose);
+        return worldTranspose;
+    }
+
+    public void updateUnityRobotBase_T_UnityRobotTCP()
+    {
+        int final = _Arms.Length - 1;
+        ArticulationBody last = _Arms[final]; // Pull the tool which is the final Articulation Body in _Arms
+        UnityRobotBase_T_UnityRobotTCP = last.transform.localToWorldMatrix; // Unity has built in function for the Matrix
+    }
+
+    public void updateRealRobotBase_T_RealRobotTCP()
```

Comment:
This should also be "Calculate" rather than "update", for similar reasons to the above function.